frodo.js

```diff
@@ -6,101 +6,90 @@
  *
```

Comment:
SCRIPT-Nice job on the comprehensive test coverage, it gives a lot of confidence in the stability of the new feature.